docs/source/cpp/compute.rst

```diff
@@ -0,0 +1,419 @@
++--------------------------+------------+----------------------------+----------------------------+
+| dictionary_encode        | Unary      | Boolean, Numeric, Temporal | Dictionary (1)             |
++--------------------------+------------+----------------------------+----------------------------+
+| dictionary_encode        | Unary      | Null                       | Dictionary (1)             |
++--------------------------+------------+----------------------------+----------------------------+
+| unique                   | Unary      | Binary- and String-like    | Input type (2)             |
++--------------------------+------------+----------------------------+----------------------------+
+| unique                   | Unary      | Boolean, Numeric, Temporal | Input type (2)             |
++--------------------------+------------+----------------------------+----------------------------+
+| unique                   | Unary      | Null                       | Input type (2)             |
++--------------------------+------------+----------------------------+----------------------------+
+| value_counts             | Unary      | Binary- and String-like    | Struct (3)                 |
++--------------------------+------------+----------------------------+----------------------------+
+| value_counts             | Unary      | Boolean, Numeric, Temporal | Struct (3)                 |
++--------------------------+------------+----------------------------+----------------------------+
+| value_counts             | Unary      | Null                       | Struct (3)                 |
++--------------------------+------------+----------------------------+----------------------------+
+
+* \(1) Output is ``Dictionary(Int32, input type)``.
+
+* \(2) Duplicates are removed from the output while the original order is
+  maintained.
+
+* \(3) Output is a ``{"values": input type, "counts": Int64}`` Struct.
+  Each output element corresponds to a unique value in the input, along
+  with the number of times this value has appeared.
+
+Selections
+~~~~~~~~~~
+
+These functions select a subset of the first input defined by the second input.
+
++-----------------+------------+---------------+--------------+------------------+-------------------------+-------------+
+| Function name   | Arity      | Input type 1  | Input type 2 | Output type      | Options class           | Notes       |
++=================+============+===============+==============+==================+=========================+=============+
+| filter          | Binary     | Any (1)       | Boolean      | Input type 1     | :struct:`FilterOptions` | \(2)        |
++-----------------+------------+---------------+--------------+------------------+-------------------------+-------------+
+| take            | Binary     | Any (1)       | Integer      | Input type 1     | :struct:`TakeOptions`   | \(3)        |
```

Comment:
These are metafunctions. Probably also want to list array_filter and array_take

Reply:
I don't know. Does it help the user to know about the non-metafunctions? Personally, I wouldn't know what to do with them.